### PR TITLE
end point  for Submitting a test

### PR DESCRIPTION
**Pull request by hnalabanda**
End point tested manually using postman
Test Cases
-https://localhost:5001/sessions/(invalid token)/ 
-https://localhost:5001/sessions/(valid token)/

### CODE REVIEW

File: TestSwitchApi/TestSwitchApi/Controllers/SessionsController.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+using Microsoft.AspNetCore.Mvc;
+using TestSwitchApi.Models.DataModels;
+using TestSwitchApi.Models.Request;
+using TestSwitchApi.Repositories;
+
+namespace TestSwitchApi.Controllers
+{
+    [ApiController]
+    [Route("/sessions")]
+    [EnableCors("AllowOrigin")]
```
Comment (MichaelWalker) [inline TestSwitchApi/TestSwitchApi/Controllers/SessionsController.cs]: shouldn't need this `EnableCors` attribute

File: TestSwitchApi/TestSwitchApi/Controllers/SessionsController.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        }
+
+        [HttpPost("{tokenId}")]
+        public ActionResult<CandidateTestModel> AddTestSubmission([FromRoute] string tokenId, [FromBody] AddTestSubmissionRequestModel newSubmission)
+        {
+            var candidate = _candidates.GetCandidateByGuid(tokenId);
+            if (candidate!=null)
+            {
+                var testSubmission = _submissions.AddTestSubmission(candidate.Id, newSubmission);
+                return testSubmission;
```
Comment (MichaelWalker) [inline TestSwitchApi/TestSwitchApi/Controllers/SessionsController.cs]: can just be `return _submissions.AddTestSubmission(candidate.Id, newSubmission);` :)

File: TestSwitchApi/TestSwitchApi/Models/Request/AddTestSubmissionRequestModel.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace TestSwitchApi.Models.Request
+{
+    public class AddTestSubmissionRequestModel
+    {
+        [Required]
+        public int TestId { get; set; }
```
Comment (MichaelWalker) [inline TestSwitchApi/TestSwitchApi/Models/Request/AddTestSubmissionRequestModel.cs:8]: we never actually check that our model complies with this `[Required]` attribute.
Could you add a check in the controller?

https://docs.microsoft.com/en-us/aspnet/core/mvc/models/validation?view=aspnetcore-3.1

**Review by MichaelWalker — CHANGES_REQUESTED** [review]
Nice! Just a couple of tiny tweeks!

File: TestSwitchApi/TestSwitchApi/Controllers/SessionsController.cs (hunk 1 above)
Comment (hnalabanda) [reply TestSwitchApi/TestSwitchApi/Controllers/SessionsController.cs]: removed EnableCors

File: TestSwitchApi/TestSwitchApi/Controllers/SessionsController.cs (hunk 2 above)
Comment (hnalabanda) [reply TestSwitchApi/TestSwitchApi/Controllers/SessionsController.cs]: corrected

File: TestSwitchApi/TestSwitchApi/Models/Request/AddTestSubmissionRequestModel.cs (hunk 3 above)
Comment (hnalabanda) [reply TestSwitchApi/TestSwitchApi/Models/Request/AddTestSubmissionRequestModel.cs:8]: Added the checks for the Required Attributes - TestId and TestAnswer

**Review by MichaelWalker — APPROVED** [review]
Nice!

**Pull request merged**